### PR TITLE
doc improvement: updated documentation for the LwM2M carrier library

### PR DESCRIPTION
**Pull request by nordic-hani**
Added some fixes and clarifications to the LwM2M carrier library usage.

Signed-off-by: Håvard Vermeer <havard.vermeer@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/libraries/bin/lwm2m_carrier/index.rst [hunk 1]
```diff
@@ -9,15 +9,18 @@ LwM2M carrier
...
 Several mobile network carriers specify their own LwM2M device management.
 The LwM2M carrier library facilitates an nRF91 device to connect to the device management servers, regardless of the other actions of the user application.
+For more information, see the list of `applicable carriers <Mobile network operator certifications_>`_, and the ref:`lwm2m_certification` subpage.
 
 .. important::
 
-   It is *mandatory* to include the LwM2M carrier library in any AT&T, T-Mobile, or Verizon certified products.
-
-For the `applicable carriers <Mobile network operator certifications_>`_, it is mandatory to include this library in your application for the certification to your end product.
+   It is *mandatory* to include the LwM2M carrier library in any Verizon certified product.
```
Comment (divipillai) [inline doc/nrf/libraries/bin/lwm2m_carrier/index.rst:16]: Shouldn't we also mention that it is optional to include the library in any AT&T and T-Mobile certified product?

File: doc/nrf/libraries/bin/lwm2m_carrier/index.rst [hunk 2]
```diff
@@ -9,15 +9,18 @@ LwM2M carrier
 
 Several mobile network carriers specify their own LwM2M device management.
 The LwM2M carrier library facilitates an nRF91 device to connect to the device management servers, regardless of the other actions of the user application.
+For more information, see the list of `applicable carriers <Mobile network operator certifications_>`_, and the ref:`lwm2m_certification` subpage.
```
Comment (umapraseeda) [inline doc/nrf/libraries/bin/lwm2m_carrier/index.rst:12]:
```suggestion
For more information, see the list of `applicable carriers <Mobile network operator certifications_>`_, and the :ref:`lwm2m_certification` subpage.
```

File: doc/nrf/libraries/bin/lwm2m_carrier/index.rst [hunk 3]
```diff
@@ -9,15 +9,18 @@ LwM2M carrier
...
 .. important::
 
-   It is *mandatory* to include the LwM2M carrier library in any AT&T, T-Mobile, or Verizon certified products.
-
-For the `applicable carriers <Mobile network operator certifications_>`_, it is mandatory to include this library in your application for the certification to your end product.
+   It is *mandatory* to include the LwM2M carrier library in any Verizon certified product.
 
 The :ref:`lwm2m_carrier` sample demonstrates how to run this library in an application.
-The :ref:`asset_tracker_v2` application also uses the LwM2M carrier library.
+In addition to this stand-alone sample, the following applications and samples include an overlay file to easily integrate the LwM2M carrier library.
```
Comment (umapraseeda) [inline doc/nrf/libraries/bin/lwm2m_carrier/index.rst:19]:
```suggestion
In addition to this stand-alone sample, the following applications and samples include an overlay file to easily integrate the LwM2M carrier library:
```

File: samples/nrf9160/lwm2m_carrier/sample_description.rst [hunk 4]
```diff
@@ -171,27 +171,34 @@ To test with the sample, complete the following steps:
...
-   #. Save the settings. Resetting the device will load them.
+   #. Apply the server config. After rebooting, the sample will now load the above settings (as opposed to using the static Kconfigs).
 
       .. code-block:: console
 
-         $ Enable custom server settings  Yes
+         $ carrier_config server enable
+         Enabled custom server config
+
+   #. Finally, as described in ref:`lwm2m_carrier_shell` we must set ``auto_startup`` (or else the sample will wait indefinetly for us to configure all the settings).
```
[the remark this answers is not in the document]
Comment (umapraseeda) [reply samples/nrf9160/lwm2m_carrier/sample_description.rst:196]: :ref:\`lwm2m_carrier_shell\`

**Review by umapraseeda — CHANGES_REQUESTED** [review]

**Review by umapraseeda — APPROVED** [review]

**Review by divipillai — APPROVED** [review]

File: samples/nrf9160/lwm2m_carrier/sample_description.rst [hunk 5]
```diff
@@ -171,27 +171,35 @@ To test with the sample, complete the following steps:
...
+   #. Apply the server config.
+      After rebooting, the sample loads these settings (instead of using the static Kconfigs).
 
       .. code-block:: console
 
-         $ Enable custom server settings  Yes
+         $ carrier_config server enable
+         Enabled custom server config
+
+   #. Finally, as described in ref:`lwm2m_carrier_shell`, we must set ``auto_startup`` (or else the sample will wait indefinitely for you to configure all the settings).
```
Comment (peknis) [inline samples/nrf9160/lwm2m_carrier/sample_description.rst:197]:
```suggestion
   #. Finally, as described in ref:`lwm2m_carrier_shell`, set ``auto_startup`` (or else the sample will wait indefinitely for you to configure all the settings).
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

**Pull request merged**